src/jit/codegencommon.cpp

```diff
@@ -7367,7 +7394,11 @@ void CodeGen::genProfilingEnterCallback(regNumber initReg, bool* pInitRegZeroed)
     // This will emit either
     // "call ip-relative 32-bit offset" or
     // "mov rax, helper addr; call rax"
-    genEmitHelperCall(CORINFO_HELP_PROF_FCN_ENTER, 0, EA_UNKNOWN);
+#if !defined(UNIX_AMD64_ABI)
+    genEmitHelperCall(CORINFO_HELP_PROF_FCN_ENTER, 0, EA_UNKNOWN, REG_ARG_2);
+#else
+    genEmitHelperCall(CORINFO_HELP_PROF_FCN_ENTER, 0, EA_UNKNOWN, REG_R11);
+#endif
```

Comment:
Is there a specific reason why we want to use REG_ARG2 on amd64 windows and REG_R11 on Unix?
Please note that without specifying a callTargetReg, genEmitHelperCall() would use REG_RAX.